packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ImageCaptureHostApiImpl.java

```diff
@@ -0,0 +1,137 @@
+import androidx.camera.core.ImageCaptureException;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.ImageCaptureHostApi;
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+
+public class ImageCaptureHostApiImpl implements ImageCaptureHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+
+  private Context context;
+  private SystemServicesFlutterApiImpl systemServicesFlutterApiImpl;
+
+  @VisibleForTesting public CameraXProxy cameraXProxy = new CameraXProxy();
+
+  public ImageCaptureHostApiImpl(
+      @NonNull BinaryMessenger binaryMessenger,
+      @NonNull InstanceManager instanceManager,
+      @NonNull Context context) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+    this.context = context;
+  }
+
+  /**
+   * Sets the context that the {@link ImageCapture} will use to find a location to save a captured
+   * image.
+   */
+  public void setContext(Context context) {
+    this.context = context;
+  }
+
+  /**
+   * Creates a {@link ImageCapture} with the requested flash mode and target resolution if
+   * specified.
+   */
+  @Override
+  public void create(
+      @NonNull Long identifier,
+      @Nullable Long flashMode,
+      @Nullable GeneratedCameraXLibrary.ResolutionInfo targetResolution) {
+    ImageCapture.Builder imageCaptureBuilder = cameraXProxy.createImageCaptureBuilder();
+    if (flashMode != null) {
+      // This sets the requested flash mode, but may fail silently.
+      imageCaptureBuilder.setFlashMode(flashMode.intValue());
+    }
+    if (targetResolution != null) {
+      imageCaptureBuilder.setTargetResolution(
+          new Size(
+              targetResolution.getWidth().intValue(), targetResolution.getHeight().intValue()));
+    }
+    ImageCapture imageCapture = imageCaptureBuilder.build();
+    instanceManager.addDartCreatedInstance(imageCapture, identifier);
+  }
+
+  /** Sets the flash mode of the {@link ImageCapture} instance with the specified identifier. */
+  @Override
+  public void setFlashMode(@NonNull Long identifier, @NonNull Long flashMode) {
+    ImageCapture imageCapture =
+        (ImageCapture) Objects.requireNonNull(instanceManager.getInstance(identifier));
+    imageCapture.setFlashMode(flashMode.intValue());
+  }
+
+  /** Captures a still image and uses the result to return its absolute path in memory. */
+  @Override
+  public void takePicture(
+      @NonNull Long identifier, @NonNull GeneratedCameraXLibrary.Result<String> result) {
+    ImageCapture imageCapture =
+        (ImageCapture) Objects.requireNonNull(instanceManager.getInstance(identifier));
+    final File outputDir = context.getCacheDir();
+    File temporaryCaptureFile;
+    try {
+      temporaryCaptureFile = File.createTempFile("CAP", ".jpg", outputDir);
```

Comment:
Consider setting these to constants and referencing them here. I think TEMP_FILE_NAME is easier to understand. I can see the argument for why you might want to hardcode .jpg but my guess is that we will actually modify that to be configurable before we are done.